### ISSUE TITLE
Change QUALIFIERS without with-redefs?

### ISSUE THREAD

**Issue by WhittlesJr** [xsc/version-clj#6]
I'd like to use this library for comparing versions of our product. Our versioning scheme uses some pretty abnormal conventions, including letters that function just like numbers essentially. So:

"PA2.06.19a-x107"  < "PA2.06.20a-x114" < "PA2.06.20b-x116" < "PA2.06.20d-x119" 

But `version-element-compare [:str :str]` references `QUALIFIERS` always, and `a` and `b` match a qualifier while `d` does not. So I get "PA2.06.20d-x119"  < "PA2.06.20b-x116".

For now I'm just going to transform those letters into numbers ahead of time before running my comparisons, but I wish there were a config map or something you could pass to the function that would allow you to customize the qualifier checking.

**Comment by WhittlesJr** [comment]
Actually I just ended up doing it this way:

```
(defn version-compare [pred val l r]
  (with-redefs [version-clj.compare/QUALIFIERS {}]
    (pred (version/version-compare l r) val)))

(def version<= (partial version-compare <= 0))
(def version>= (partial version-compare >= 0))
```

**Comment by WhittlesJr** [comment]
So I guess the issue is just that redef'ing a private var feels wrong.

**Comment by gfredericks** [comment]
the biggest reason it's "wrong" is because it's not threadsafe

**Comment by xsc** [comment]
I just released version `2.0.0` that allows you to pass a `{:qualifiers { ... }}` map to all of version-clj's functions (e.g. [`version-compare`](https://cljdoc.org/d/version-clj/version-clj/2.0.0/api/version-clj.core#version-compare)). Probably a few years too late, apologies. :)
